src/ansi-c/library/string.c

```diff
@@ -374,8 +374,11 @@ inline int strcmp(const char *s1, const char *s2)
   unsigned char ch1, ch2;
   do
   {
+#  pragma CPROVER check push
```

Comment:
Why are these pragmas necessary? It seems to be the problem is rather that `ch1` and `ch2` are declared as `unsigned char` rather than `char` (which may or may not be `unsigned` depending on configuration).

Reply:
@hannes-steffenhagen-diffblue Yep, the source of the problem is that `ch1` and `ch2` are declared as `unsigned char`; however, `\xff` doesn't work correctly with `signed char` (see the test case `String6`). Thus, we can either fix that (special) case or use `unsigned char` and add `#pragma CPROVER check disable "conversion"`. For me, the latter makes sense because the standard for these functions interpreted this characters as type `unsigned char`, which I mentioned at the description of this PR.